lib/s3_inventory.rb

```diff
@@ -4,16 +4,23 @@
 require "csv"
 
 class S3Inventory
-  attr_reader :type, :model, :inventory_date
+  attr_reader :type, :model, :inventory_date, :s3_helper
 
   CSV_KEY_INDEX ||= 1
   CSV_ETAG_INDEX ||= 2
   INVENTORY_PREFIX ||= "inventory"
   INVENTORY_VERSION ||= "1"
   INVENTORY_LAG ||= 2.days
 
-  def initialize(s3_helper, type, preloaded_inventory_file: nil, preloaded_inventory_date: nil)
```

Comment:
I removed `s3_helper` from the arguments accepted by the constructor because the only valid instances of `S3Helper` are those generated in this constructor.